src/viur/core/modules/file.py

```diff
@@ -677,22 +695,26 @@ def read(self, key: db.Key | int | str | None = None, path: str | None = None) -
 
         :param key: Key of the LeafSkel that contains the "dlkey" and the "name".
         :param path: The path of the file in the Cloud Storage Bucket.
+        :param public: Defines whether the file path should be taken from private or public repo.
 
         :return: Returns the file as a io.BytesIO buffer and the content-type
         """
-        current_bucket = GOOGLE_STORAGE_BUCKET
         if not key and not path:
             raise ValueError("Please provide a key or a path")
+
         if key:
             skel = self.viewSkel("leaf")
             if not skel.fromDB(db.keyHelper(key, skel.kindName)):
                 if not path:
                     raise ValueError("This skeleton is not in the database!")
             else:
                 path = f"""{skel["dlkey"]}/source/{skel["name"]}"""
-            current_bucket = get_current_bucket(skel["dlkey"])
 
-        blob = current_bucket.blob(path)
+            bucket = File.get_bucket(skel["dlkey"])
+        else:
+            bucket = File.get_bucket(PUBLIC_DLKEY_POSTFIX if public else "")
```

Comment:
This will not work because the prefix is not a dlkey. The public parameter is not necessary, as the type is defined by the dlkey in both the skel and the path

Reply:
Can you clarify with a suggestion? In case there is only a path provided, there is still no way to define whether the path belongs to the public or private bucket.